Source/WebKit/UIProcess/API/Cocoa/_WKWebExtensionContext.h

```diff
@@ -453,6 +453,119 @@ WK_CLASS_AVAILABLE(macos(13.3), ios(16.4))
+ @param focusedWindow The window that gained focus, or \c nil if no window has focus or a window has focus that is not visible to this extension.
+ @discussion This method informs only the specific extension that a window has gained focus. If the intention is to inform all loaded
+ extensions consistently, you should use the respective method on the extension controller instead.
+ */
+- (void)didFocusWindow:(nullable id <_WKWebExtensionWindow>)focusedWindow;
+
+/*!
+ @abstract Should be called by the app when a new tab is opened to fire appropriate events with only this extension.
+ @param newTab The newly opened tab.
+ @discussion This method informs only the specific extension of the opening of a new tab. If the intention is to inform all loaded
+ extensions consistently, you should use the respective method on the extension controller instead.
+ @seealso didCloseTab:
+ @seealso openTabs
+ */
+- (void)didOpenTab:(id <_WKWebExtensionTab>)newTab;
+
+/*!
+ @abstract Should be called by the app when a tab is closed to fire appropriate events with only this extension.
+ @param closedTab The tab that was closed.
+ @param windowIsClosing A boolean value indicating whether the window containing the tab is also closing.
+ @discussion This method informs only the specific extension of the closure of a tab. If the intention is to inform all loaded 
+ extensions consistently, you should use the respective method on the extension controller instead.
+ @seealso didOpenTab:
+ @seealso openTabs
+ */
+- (void)didCloseTab:(id <_WKWebExtensionTab>)closedTab windowIsClosing:(BOOL)windowIsClosing;
+
+/*!
+ @abstract Should be called by the app when tabs are selected to fire appropriate events with only this extension.
+ @param selectedTabs The set of tabs that were selected. An empty set indicates that no tabs are currently selected or that the
+ selected tabs are not visible to this extension.
+ @discussion This method informs only the specific extension that tabs have been selected. If the intention is to inform all loaded
+ extensions consistently, you should use the respective method on the extension controller instead.
+ */
+- (void)didSelectTabs:(NSSet<id <_WKWebExtensionTab>> *)selectedTabs;
+
+/*!
+ @abstract Should be called by the app when a tab is moved to fire appropriate events with only this extension.
+ @param movedTab The tab that was moved.
+ @param index The old index of the tab within the window.
+ @param oldWindow The window that the tab was moved from, or \c nil if the window stayed the same.
+ @discussion This method informs only the specific extension that a tab has been moved. If the intention is to inform all loaded 
+ extensions consistently, you should use the respective method on the extension controller instead.
+ */
+- (void)didMoveTab:(id <_WKWebExtensionTab>)movedTab fromIndex:(NSUInteger)index inWindow:(nullable id <_WKWebExtensionWindow>)oldWindow NS_SWIFT_NAME(didMoveTab(_:from:in:));
```

Comment:
How will we get the new index of the tab in the window?

Reply:
Will ask the `movedTab`'s `window`'s `tabs` array for the current index of the object. It should be moved in the array before this call.